domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/AppHealthMetricController.kt

```diff
@@ -0,0 +1,248 @@
+package org.oppia.android.domain.oppialogger.analytics
+
+import org.oppia.android.app.model.OppiaMetricLog
+import org.oppia.android.app.model.OppiaMetricLog.Priority
+import org.oppia.android.app.model.OppiaMetricLogs
+import org.oppia.android.data.persistence.PersistentCacheStore
+import org.oppia.android.domain.oppialogger.MetricLogStorageCacheSize
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.logging.ConsoleLogger
+import org.oppia.android.util.logging.ExceptionLogger
+import org.oppia.android.util.logging.performancemetrics.PerformanceMetricLogger
+import org.oppia.android.util.metriccollection.AppHealthMetricUtils
+import org.oppia.android.util.metriccollection.AppStartupMetrics
+import org.oppia.android.util.metriccollection.UiSpecificMetrics
+import org.oppia.android.util.networking.NetworkConnectionUtil
+import java.lang.IllegalStateException
+import javax.inject.Inject
+
+/**
+ * Controller for handling performance metrics event logging.
+ *
+ * Callers should not use this class directly; instead, they should use ``OppiaLogger`` which
+ * provides convenience log methods.
+ */
+class AppHealthMetricController @Inject constructor(
```

Comment:
Let's keep this specific to perfornance metrics. Health metrics are a broader category that will include non-performance metrics in the future.

Reply:
Done.